front/src/modules/simulationResult/components/SpeedSpaceChart/helpers.ts

```diff
@@ -14,10 +14,75 @@ import type {
   SimulationResponseSuccess,
 } from 'applications/operationalStudies/types';
 import type { ReportTrainV2 } from 'common/api/osrdEditoastApi';
-import { mmToKm, msToKmh, mToKm } from 'utils/physics';
+import { mmToKm, mmToM, msToKmh, mToKm } from 'utils/physics';
 
 import { electricalProfilesDesignValues } from './consts';
 
+type SpeedLimitTagInput = {
+  boundaries: number[];
+  values: {
+    speed: number;
+    source: { speed_limit_source_type: string; tag?: string } | null;
+  }[];
+};
+
+const getTagName = (source: { speed_limit_source_type: string; tag?: string } | null): string => {
```

Comment:
- You can use the real type from editoast ;)
- then you will have to remove the extra `''` (in each return), check that source is not undefined and remove the impossible case

```suggestion
type SpeedLimitTagValue = ArrayElement<SimulationResponseSuccess['mrsp']['values']>;

const getTagName = (source: SpeedLimitTagValue['source']): string => {
```